tests/unit/src/test_conditionsStore.cpp

```diff
@@ -663,6 +663,49 @@ SCENARIO( "Providing multiple derived conditions", "[ConditionStore][DerivedMult
 }
 
 
+
+SCENARIO("Using expanded conditions", "[ConditionsStore][Expand]")
+{
+	GIVEN("A ConditionsStore with existing values")
+	{
+		auto store = ConditionsStore();
+		store["test"] = 10;
+		THEN("Conditions are expanded")
+		{
+			REQUIRE(store.Set("test$[test]test", 1));
+			REQUIRE(store.Get("test$[test]test") == 1);
+			REQUIRE(store.Get("test10test") == 1);
+			REQUIRE(store.PrimariesSize() == 2);
+			REQUIRE( store["test10test"] == 1 );
+			REQUIRE( store["test$[test]test"] == 1 );
+		}
+		AND_THEN("Multiple and nested expansions are possible")
+		{
+			REQUIRE( store.Set("$[test]a", 2) );
+			REQUIRE( store.Set("a$[test]", 3) );
+			REQUIRE( store.Set("$[test]", 4) );
+			REQUIRE( store.Get("10a") == 2);
+			REQUIRE( store.Get("a10") == 3);
+			REQUIRE( store.Get("10") == 4);
+			REQUIRE( store.Set("hello$[$[test]]", 5) );
+			REQUIRE( store.Get("hello4") == 5 );
+			REQUIRE( store.Set("$[test]$[test]", 100) );
+			REQUIRE( store.Set("1010", 100) );
+		}
+		THEN("Invalid expansions are ignored")
+		{
+			REQUIRE( store.Set("hello$[test", 1) );
+			REQUIRE( store.Get("hello$[test") == 1 );
+		}
+		THEN("Unbalanced expansions are handled")
+		{
+			REQUIRE( store.Set("hi$[test]]", 1) );
+			REQUIRE( store.Get("hi10]") == 1 );
+		}
+	}
```

Comment:
Should we also test what happens when people start adding 1 (in 10 steps) to the following condition; `$[test $[test $[test 4]]]` ? What do we actually expect to happen?

Reply:
> `$[test $[test $[test 4]]]`

This is not actually a problem, since we aren't modifying the condition during lookup. The logic is something like this:

1) Found an `$[`, expanding `test $[test $[test 4]]`
    1.1) string is `test `, found `$[`, expanding `test $[test 4]`
        1.1.1) string is `test `, found `$[`, expanding `test 4`
           1.1.1.1) string is `test 4`, its value is (for instance) `0`.
        1.1.2) string is `test 0`, its value is `0`
    1.2) string is `test 0`, its value is `0`
2. So this is equivalent to looking up `0`, which probably has the value `0`.

If we increment this value, we increment the `0` condition.